File: linkedlist.py
class Node:
  def __init__(self,data):
    self.data=data
    self.next=None
class Singly_linked:
  def __init__(self):
    self.head=None
    self.tail=None

  def insert_beg(self,data):
    if(self.head==None):
      newnode=Node(data)
      self.head=newnode
      self.tail=newnode
    else:
      newnode=Node(data)
      newnode.next=self.head
      self.head=newnode
    newnode=None

  def insert_end(self,data):
    if(self.head==None):
      newnode=Node(data)
      self.head=newnode
      self.tail=newnode
    else:
      newnode=Node(data)
      self.tail.next=newnode
      self.tail=self.tail.next
    newnode=None
  
  def insert_pos(self,data,pos):
    if(self.head==None):
      newnode=Node(data)
      self.head=newnode
      self.tail=newnode
    else:
      newnode=Node(data)
      temp=self.head
      for i in range(1,pos-1):
        temp=temp.next
      newnode.next=temp.next
      temp.next=newnode
    newnode=None
  
  def delete_beg(self):
    temp=self.head
    self.head=temp.next
    temp.next=None

  def delete_end(self):
    temp=self.head
    while(temp.next.next!=None):
      temp=temp.next
    temp.next=None
  
  def delete_pos(self,pos):
    temp=self.head
    for i in range(1,pos-1):
      temp=temp.next
    new=temp.next
    temp.next=new.next
    new.next=None
  
  def search(self,x):
    temp=self.head
    while(temp!=None):
      if(temp.data==x):
        return True
      temp=temp.next
    return False
   
  def le(self):
    if(self.head==None):
      return 0
    temp=self.head
    count=0
    while(temp!=None):
      count=count+1
      temp=temp.next
    return count

  def display(self):
    temp=self.head
    while(temp.next!=None):
      print(temp.data,end='->')
      temp=temp.next
    print(temp.data)

def menu():
  print("Linkedlist")
  s=Singly_linked()
  while(1):
    print("1-->insert")
    print("2-->delete")
    print("3-->search")
    print("4-->display")
    print("5-->count number of elements")
    print("6-->exit")
    x=int(input())
    if(x==1):
      print("1--> insert at beginning")
      print("2--> insert at end")
      print("3--> insert in between beginning and end")
      y=int(input())
      val=input("data to insert:")
      if(y==1):
        s.insert_beg(val)
      if(y==2):
        s.insert_end(val)
      if(y==3):
        b=int(input("position:"))
        s.insert_pos(val,b)
    elif(x==2):
      if(s.le()>0):
        print("1--> delete at beginning")
        print("2--> delete at end")
        print("3--> delete in between beginning and end")
        y=int(input())
        if(y==1):
          s.delete_beg()
        if(y==2):
          s.delete_end()
        if(y==3):
          b=int(input("position:"))
          s.delete_pos(b)
      else:
        print("no elements found")
    elif(x==3):
      if(s.le()>0):
        a=input("data:")
        b=s.search(a)
        if(b):
          print("yes")
        else:
          print("No")
      else:
        print("no elements found")
    elif(x==5):
      print("count",s.le())
    elif(x==4):
      if(s.le()>0):
        s.display()
      else:
        print("no elemenys found")
    elif(x==6):
      break

menu()

File: Linkedlist.java
import java.util.*;
class Node
{
  int data;
  Node next;
  Node(int d)
  {
    data=d;
    next=null;
  }
}

class Linkedlist
{
  Node head;
  Node tail;

  void insert_beg(Node n)
  {
    if(head==null)
    {
      head=n;
      tail=n;
      head.next=null;
    }
    else
    {
      n.next=head;
      head=n;
    }
    n=null;
  }

  void insert_pos(Node n,int p)
  {
    if(head==null)
    {
      head=n;
      tail=n;
      head.next=null;
    }
    else
    {
      Node temp=head;
      for(int i=1;i<p-1;i++)
      {
        temp=temp.next;
      }
      n.next=temp.next;
      temp.next=n;
    }
    n=null;
  }

  void insert_end(Node n)
  {
    if(head==null)
    {
      head=n;
      tail=n;
      head.next=null;
    }
    else
    {
      n.next=null;
      tail.next=n;
      tail=tail.next;
    }
    n=null;
  }

  void del_beg()
  {
    Node temp=head;
    head=head.next;
    temp.next=null;
  }

  void del_end()
  {
    Node temp=head;
    while(temp.next.next!=null)
    {
      temp=temp.next;
    }
    temp.next=null;
    tail=temp;
  }

  void del_pos(int p)
  {
    Node temp=head;
    for(int i=1;i<p-1;i++)
    {
      temp=temp.next;
    }
    Node new1=temp.next;
    temp.next=new1.next;
    new1.next=null;
  }

  void display()
  {
    Node temp=head;
    while(temp.next!=null)
    {
      System.out.print(temp.data+"-->");
      temp=temp.next;
    }
    System.out.println(temp.data);
  }

  int search(int x)
  {
    Node temp=head;
    while(temp!=null)
    {
      if(temp.data==x)
      {
        return 1;
      }
      temp=temp.next;
    }
    return 0;
  }

  public static void main(String args[])
  {
    Scanner sc=new Scanner(System.in);
    Linkedlist l=new Linkedlist();
    while(true)
    {
      System.out.println("1-->insert");
      System.out.println("2-->delete");
      System.out.println("3-->display");
      System.out.println("4-->search");
      int x=sc.nextInt();
      if(x==1)
      {
        System.out.println("1-->beginning");
        System.out.println("2-->end");
        System.out.println("3-->position");
        int y=sc.nextInt();
        System.out.println("enter data:");
        int data=sc.nextInt();
        Node newnode=new Node(data);
        if(y==1)
        {
          l.insert_beg(newnode);
        }
        if(y==2)
        {
          l.insert_end(newnode);
        }
        if(y==3)
        {
          System.out.println("enter position to insert:");
          int p=sc.nextInt();
          l.insert_pos(newnode,p);
        }
      }
      else if(x==2)
      {
        System.out.println("1-->beginning");
        System.out.println("2-->end");
        System.out.println("3-->position");
        int z=sc.nextInt();
        if(z==1)
        {
          l.del_beg();
        }
        if(z==2)
        {
          l.del_end();
        }
        if(z==3)
        {
          int p=sc.nextInt();
          l.del_pos(p);
        }
      }
      else if(x==3)
      {
        l.display();
      }
      else if(x==4)
      {
        int a=sc.nextInt();
        int s=l.search(a);
        if(s==1)
        {
          System.out.println("yes");
        }
        else
        {
          System.out.println("no");
        }
      }
    }
  }
}
